integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/nextmove/v2/ContentDisposition.java

```diff
@@ -0,0 +1,261 @@
+        }
+        if (this.filename != null) {
+            if (this.charset == null || StandardCharsets.US_ASCII.equals(this.charset)) {
+                sb.append("; filename=\"");
+                sb.append(this.filename).append('\"');
+            } else {
+                sb.append("; filename*=");
+                sb.append(encodeHeaderFieldParam(this.filename, this.charset));
+            }
+        }
+        if (this.size != null) {
+            sb.append("; size=");
+            sb.append(this.size);
+        }
+        if (this.creationDate != null) {
+            sb.append("; creation-date=\"");
+            sb.append(RFC_1123_DATE_TIME.format(this.creationDate));
+            sb.append('\"');
+        }
+        if (this.modificationDate != null) {
+            sb.append("; modification-date=\"");
+            sb.append(RFC_1123_DATE_TIME.format(this.modificationDate));
+            sb.append('\"');
+        }
+        if (this.readDate != null) {
+            sb.append("; read-date=\"");
+            sb.append(RFC_1123_DATE_TIME.format(this.readDate));
+            sb.append('\"');
+        }
+        return sb.toString();
+    }
+
+    /**
+     * Return an empty content disposition.
+     */
+    public static ContentDisposition empty() {
+        return new ContentDisposition("", null, null, null, null, null, null, null);
+    }
+
+    /**
+     * Parse a {@literal Content-Disposition} header value as defined in RFC 2183.
+     *
+     * @param contentDisposition the {@literal Content-Disposition} header value
+     * @return the parsed content disposition
+     * @see #toString()
+     */
+    public static ContentDisposition parse(String contentDisposition) {
+        List<String> parts = tokenize(contentDisposition);
+        String type = parts.get(0);
+        String name = null;
+        String filename = null;
+        Charset charset = null;
+        Long size = null;
+        OffsetDateTime creationDate = null;
+        OffsetDateTime modificationDate = null;
+        OffsetDateTime readDate = null;
+        for (int i = 1; i < parts.size(); i++) {
+            String part = parts.get(i);
+            int eqIndex = part.indexOf('=');
+            if (eqIndex != -1) {
+                String attribute = part.substring(0, eqIndex);
+                String value = (part.startsWith("\"", eqIndex + 1) && part.endsWith("\"") ?
+                        part.substring(eqIndex + 2, part.length() - 1) :
+                        part.substring(eqIndex + 1, part.length()));
+                if (attribute.equals("name")) {
+                    name = value;
+                } else if (attribute.equals("filename*")) {
+                    filename = decodeHeaderFieldParam(value);
+                    charset = Charset.forName(value.substring(0, value.indexOf('\'')));
+                    Assert.isTrue(UTF_8.equals(charset) || ISO_8859_1.equals(charset),
+                            "Charset should be UTF-8 or ISO-8859-1");
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "Charset should be UTF-8 or ISO-8859-1" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1192)